app/lib/csv/exporter.rb

```diff
@@ -44,10 +44,8 @@ def to_send_data
     [ to_csv, send_options ]
   end
 
-  def generate
-    ::CSV.generate(csv_options) do |csv|
-      yield csv
-    end
+  def generate(&block)
+    ::CSV.generate(**csv_options, &block)
```

Comment:
Original seems ok. Using more performant approach and requires a block to be given.

Reply:
You mean
```
  def generate
    ::CSV.generate(**csv_options) do |csv|
      yield csv
    end
  end
``` 
? (because we still need to convert `csv_options` to kwargs)

Passing `&block` was suggested by RuboCop:
![image](https://github.com/user-attachments/assets/4da2d341-36fb-4331-91fc-0a31a738d260)

https://docs.rubocop.org/rubocop/cops_style.html#styleexplicitblockargument

Reply:
Now `generate` can be called without a block which would fail previously :thinking: 
I don't like allowing things that were not allowed previously although this looks pretty innocent.

Reply:
As`::CSV.generate` expects a block, it would still need to be passed.

And apart from that, I've been playing around with these examples:

```
def call
  puts 'before block'
  yield
  puts 'after block'
end

def generate1(&block)
  call(&block)
end

generate1 { puts 'hello' }
generate1

```

The outcome of the last one is:
```
irb(main):012:0> generate1
before block
(irb):3:in `call': no block given (yield) (LocalJumpError)
        from (irb):8:in `generate1'  
        from (irb):12:in `<main>'   
```

If compared to:

```
def generate2
  call do
    yield
  end
end

generate2 { puts 'hello' }
generate2
```

The result of the last call is:
```
irb(main):021:0> generate2
before block
(irb):15:in `block in generate2': no block given (yield) (LocalJumpError)
        from (irb):3:in `call'                               
        from (irb):14:in `generate2'                         
        from (irb):21:in `<main>'  
```

Both fail, and in fact it looks like `generate1` (the "new" syntax) even fails earlier? Or maybe I'm wrong, or maybe the example is not relevant.

In any case, to me the new syntax is much more clear - you see right away that the method expects a block, and also I think it's easier to follow the flow.

Reply:
if `::CSV.generate` requires a block, then that should be no difference in behavior. Historically `yield` was much faster than passing blocks but readability is more important outside perf critical path.